nvflare/apis/impl/controller.py

```diff
@@ -628,7 +616,8 @@ def abort_task(self, task, fl_ctx: FLContext):
 
     def _end_task(self, task_names, fl_ctx: FLContext):
         engine = fl_ctx.get_engine()
-        assert isinstance(engine, ServerEngineSpec)
+        if not isinstance(engine, ServerEngineSpec):
+            raise TypeError("engine should be an instance of ServerEngineSpec, but get {}".format(type(engine)))
```

Comment:
"get" should be "got" here since the error is being raised after the wrong engine type was gotten

Reply:
fixed